ReSharper.FSharp/src/FSharp.Psi.Features/src/QuickFixes/ConvertTupleToElementsFix.fs

```diff
@@ -0,0 +1,76 @@
+﻿namespace JetBrains.ReSharper.Plugins.FSharp.Psi.Features.Daemon.QuickFixes
+
+open System.Text.RegularExpressions
+open JetBrains.ReSharper.Feature.Services.Intentions.Scoped
+open JetBrains.ReSharper.Feature.Services.Intentions.Scoped.Actions
+open JetBrains.ReSharper.Feature.Services.Intentions.Scoped.Scopes
+open JetBrains.ReSharper.Plugins.FSharp.Psi
+open JetBrains.ReSharper.Plugins.FSharp.Psi.Features.Daemon.Highlightings
+open JetBrains.ReSharper.Plugins.FSharp.Psi.Features.Daemon.QuickFixes
+open JetBrains.ReSharper.Plugins.FSharp.Psi.Impl
+open JetBrains.ReSharper.Plugins.FSharp.Psi.Tree
+open JetBrains.ReSharper.Psi.ExtensionsAPI
+open JetBrains.ReSharper.Psi.ExtensionsAPI.Tree
+open JetBrains.ReSharper.Psi.Util
+open JetBrains.ReSharper.Resources.Shell
+
+module ConvertTupleToElements =
+    let regex = Regex("'\w\d*( \* '\w\d*)+")
+
+    let getArrayOrListParent (expr: IFSharpExpression) =
+        let expr = expr.IgnoreParentParens()
+        let arrayOrList = ArrayOrListExprNavigator.GetByExpression(expr)
+        let arrayOrList =
+            if isNotNull arrayOrList then arrayOrList else
+                let seq = SequentialExprNavigator.GetByExpression(expr)
+                if isNotNull seq then ArrayOrListExprNavigator.GetByExpression(seq) else null
+        arrayOrList
+
+    let isAvailable expr actual =
+        isValid expr && isNotNull (getArrayOrListParent expr) && regex.Match(actual).Value = actual
+
+    let convertTupleToElements (expr: IFSharpExpression) (factory: IFSharpElementFactory) =
+        match expr with
+        | :? ITupleExpr as tuple ->
+            let sequentialExpr = factory.CreateSequentialExpr(tuple.ExpressionsEnumerable)
+            ModificationUtil.ReplaceChildRange(
+                TreeRange(expr.IgnoreParentParens()),
+                TreeRange(sequentialExpr.FirstChild, sequentialExpr.LastChild)) |> ignore
+        | _ -> ()
+
+type ConvertTupleToElementsFix(warning: UnexpectedExpressionTypeError) =
+    inherit FSharpQuickFixBase()
+
+    let actual = warning.ActualType
+    let expr = warning.Expr
+    let parent = ConvertTupleToElements.getArrayOrListParent expr
+
+    override x.Text =
+        if parent.IsArray then "Convert tuple to array elements" else "Convert tuple to list elements"
```

Comment:
`parent` sounds like `ITreeNode.Parent: ITreeNode` and _apparently_ it has a different type. Could it be something like `arrayOrListExpr` instead, giving a hint about its type?

Reply:
forgot to rename before pushing, thanks